src/algorithms/dqns/common.jl

```diff
@@ -35,3 +35,42 @@ function RLBase.update!(
     push!(trajectory[:terminal], is_terminated(env))
     push!(trajectory[:priority], p.learner.default_priority)
 end
+
+#####
+# some common components for Dueling network
+#####
+
+struct DuelingNetwork
+    base::Chain
+    val::Chain
+    adv::Chain
+end
+
+function build_dueling_network(network::Chain, device, n_atoms::Int=1)
```

Comment:
This function is over specific here.

1. `n_atoms` makes this function specific to C51 (Rainbow)
2. `device` is not needed here. Since you've already applied the `Flux.@functor` to the `DuelingNetwork` structure. Users should be responsible for where to store the model.
3. The `val` and `adv` do not share the `base` part. However, sometimes this is intended.

I prefer to move this function in `Dopamine_Rainbow_Atari.jl` if you still think this function is useful.

Reply:
I think it would be better to move this function to `rainbow.jl`.